src/aggregates/gapfill_delta.rs

```diff
@@ -1,8 +1,31 @@
 use crate::aggregates::{Microseconds, Milliseconds, STALE_NAN, USECS_PER_MS, USECS_PER_SEC};
+use crate::palloc::{Inner, InternalAsValue};
 use pgx::*;
 use serde::{Deserialize, Serialize};
 use std::collections::VecDeque;
 
+#[pg_extern(immutable, parallel_safe)]
+pub fn prom_extrapolate_final(state: Internal) -> Option<Vec<Option<f64>>> {
+    prom_extrapolate_final_inner(unsafe { state.to_inner() })
+}
+pub fn prom_extrapolate_final_inner(
+    state: Option<Inner<GapfillDeltaTransition>>,
+) -> Option<Vec<Option<f64>>> {
+    state.map(|mut s| s.as_vec())
+}
+
+/// Backwards compatibility
+#[no_mangle]
+pub extern "C" fn pg_finfo_gapfill_delta_final() -> &'static pg_sys::Pg_finfo_record {
+    const V1_API: pg_sys::Pg_finfo_record = pg_sys::Pg_finfo_record { api_version: 1 };
+    &V1_API
+}
```

Comment:
should we consider making this a macro?

Reply:
I thought about it, but I didn't think it would be worth it for code that we would probably drop when we have versioned SOs anyway.